src/lock.js

```diff
@@ -68,4 +69,42 @@ class Lock {
   }
 }
 
-export const lock = new Lock();
+class GlobalLock {
+  constructor() {
+    this.wallets = {};
```

Comment:
suggestion (blocking): We should add a comment explaining that this is to store a lock per wallet where the key is the wallet-id and the value is the `Lock` object.

Reply:
Done